bridge/engine/EngineStreamDirector.h

```diff
@@ -460,30 +472,29 @@ class EngineStreamDirector
 
         const auto pinMapItr = _pinMap.find(toEndpointIdHash);
         const bool fromPinnedEndpoint = pinMapItr != _pinMap.end() && isSsrcFromParticipant(pinMapItr->second, ssrc);
-        const auto maxWantedQuality =
-            (fromPinnedEndpoint ? viewer->desiredHighestEstimatedPinnedLevel : viewer->desiredUnpinnedLevel);
+        const auto assignedQuality = (fromPinnedEndpoint ? viewer->pinQualityLevel : viewer->unpinQualityLevel);
```

Comment:
This was the main bug. We used desired quality to figure out if ssrc is used, not the actual selected quality